### PR TITLE
[Update] Firebase for iOS 6.21.0

### LINKED ISSUE #319: Xamarin IOS build error
I get this error on build:
```
Severity	Code	Description	Project	File	Line	Suppression State
Error		Could not copy the file "GCnsnt-1.0.3\googleads-consent-sdk-ios-1.0.3\PersonalizedAdConsent\PersonalizedAdConsent\PersonalizedAdConsent.bundle\consentform.html" because it was not found.	 Platform.iOS			
```

The error appears after upgrade Xamarin.Google.iOS.MobileAds from 7.38 to 7.47

### LINKED ISSUE #369: Includes Deprecated API UIWebView In Firebase ML
Using MLKit it includes the API UIWebView, Which is a warning but in April  2020 Apple will stop accepting submissions with UIWebVew for new Apps. Existing Apps will stop accepting in December 2020  

ITMS-90809: Deprecated API Usage - Apple will stop accepting submissions of apps that use UIWebView APIs . See https://developer.apple.com/documentation/uikit/uiwebview for more information.

Sample
[TestMLUpgrade.zip](https://github.com/xamarin/GoogleApisForiOSComponents/files/4214024/TestMLUpgrade.zip)

I cannot find any usage in the Headers but doing strings on file i can see it.

```
Davids-Mac-mini:TestML.app davidlilley$ pwd
/Users/xxx/Projects/TestMLUpgrade/TestML/bin/iPhone/Release/TestML.app

Davids-Mac-mini:TestML.app xx$ strings TestML | grep UIWebView
UIWebViewDelegate
c20@0:4@"UIWebView"8@"NSURLRequest"12i16
v12@0:4@"UIWebView"8
v16@0:4@"UIWebView"8@"NSError"12
@"UIWebView"
UIWebViewDelegate
B40@0:8@"UIWebView"16@"NSURLRequest"24q32
v24@0:8@"UIWebView"16
v32@0:8@"UIWebView"16@"NSError"24
@"UIWebView"

```

### LINKED ISSUE #373: dyld: Library not loaded: @rpath/protobuf.framework/protobuf
After upgrading my nuget packages to Xamarin.Firebase.Auth 6.4.2, I am seeing a runtime error similar to [#322](https://github.com/xamarin/GoogleApisForiOSComponents/issues/322) 
I am using Xamarin.Firebase.iOS.CloudMessaging 4.1.1.1 

I get the error on app launch
`dyld: dependent dylib '@rpath/Protobuf.framework/Protobuf' not found for '/private/var/containers/Bundle/Application/9C77982B-C2D2-448B-8482-B012516643F4/myAppiOS.app/Frameworks/FirebaseMessaging.framework/FirebaseMessaging', tried but didn't find: '/private/var/containers/Bundle/Application/9C77982B-C2D2-448B-8482-B012516643F4/myAppiOS.app/Frameworks/Protobuf.framework/Protobuf' '/private/var/containers/Bundle/Application/9C77982B-C2D2-448B-8482-B012516643F4/myAppiOS.app/Frameworks/FirebaseMessaging.framework/Frameworks/Protobuf.framework/Protobuf' '/private/var/containers/Bundle/Application/9C77982B-C2D2-448B-8482-B012516643F4/myAppiOS.app/Frameworks/Protobuf.framework/Protobuf' '@rpath/Protobuf.framework/Protobuf' '/System/Library/Frameworks/Protobuf.framework/Protobuf'`

Prior to upgrading, I was using these versions:
**Xamarin.Firebase.iOS.Core 6.1.0
Xamarin.Firebase.iOS.Auth 6.2.1.1**
Xamarin.Firebase.iOS.CloudMessaging 4.1.1.1 
Xamarin.Firebase.iOS.InstanceID 4.2.1
Xamarin.Firebase.iOS.Database 6.0.0

After upgrading nuget packages I am using 
**Xamarin.Firebase.iOS.Core 6.6.0
Xamarin.Firebase.iOS.Auth 6.4.2**
Xamarin.Firebase.iOS.CloudMessaging 4.1.1.1 
Xamarin.Firebase.iOS.InstanceID 4.2.1
Xamarin.Firebase.iOS.Database 6.0.0

Is there anything wrong with my current configuration?

### LINKED ISSUE #375: Dynamic links 4.0.1.1 still referencing UIWebView
I'm getting this warning when submitting my apps to Apple:

 ITMS-90809: Deprecated API Usage - Apple will stop accepting submissions of apps that use UIWebView APIs starting from April 2020 . See https://developer.apple.com/documentation/uikit/uiwebview for more information.

If I remove my reference to Xamarin.Firebase.iOS.DynamicLinks I don't get the warning any more.

If I run this command: strings events.app | grep UIWebView

I get this output only when the dynamic links reference is in place:
UIWebViewDelegate
B40@0:8@"UIWebView"16@"NSURLRequest"24q32
v24@0:8@"UIWebView"16
v32@0:8@"UIWebView"16@"NSError"24
@"UIWebView"

This is getting urgent because Apple say the deadline for new apps is 'april': https://developer.apple.com/news/?id=12232019b

Thanks!

### LINKED ISSUE #379: [Bug] Xamarin.Firebase.iOS.CloudFirestore not working with the new version of Xamarin.Firebase.iOS.Core
Description
After adding Xamarin.Firebase.iOS.CloudFirestore package in my new Xamarin.iOS project with all its dependencies, app fails to launch after successful builds. I noticed after removing all references to Firestore in the code, the app would launch as normal.. After further playing around with the packages, downgrading Xamarin.Firebase.iOS.Core to 6.1.0.1 seems to be working for Firestore but breaks other Firebase packages that depend on Xamarin.Firebase.iOS.Core to 6.6.0

Steps to Reproduce
1. Start a new project Xamarin iOS solution or use an existing project 
2. Add Xamarin.Firebase.iOS.CloudFirestore package to the solution and update to the latest stable version
3.  Ensure Xamarin.Build.Download and Xamarin.Firebase.iOS.Core are updated

Expected Behavior
- application should build and deploy to simulator of device

Actual Behavior
- application builds but exit on attempt to deploy

Basic Information
- Version with issue: 6.6.0
- Last known good version: Xamarin.Firebase.iOS.Core 6.1.0.1
- IDE: visual studio for Mac
- Platform Target Frameworks: 
  - iOS:  
- Nuget Packages: Xamarin.Firebase.iOS.Core 6.6.0
- Affected Devices: iOS

### LINKED ISSUE #380: Update Google MobileAds to use googleads-consent-sdk-ios 1.0.5 to remove UIWebView
googleads-consent-sdk-ios has been updated from 1.0.4 to 1.0.5 to remove UIWebView:

https://github.com/googleads/googleads-consent-sdk-ios/releases/tag/v1.0.5
https://github.com/googleads/googleads-consent-sdk-ios/commit/5cc10860a9e3edec73264c5fe8a77759b8d11148
https://github.com/googleads/googleads-consent-sdk-ios/issues/41

The Google MobileAds dependency now needs to be updated to this version to remove support for the deprecated UIWebView.

### LINKED ISSUE #382: MLVison Sample Proj shows " Failed to locate required resources. Please verify the resource bundle is imported.
Using the latest packages ML Kit fails to do text recognition.
In the Application output I get

"Failed to locate required resources. Please verify the resource bundle is imported."

Latest Nuggets are 
 <package id="Xamarin.Build.Download" version="0.10.0" targetFramework="xamarinios10" />
  <package id="Xamarin.Firebase.iOS.Core" version="6.6.0" targetFramework="xamarinios10" />
  <package id="Xamarin.Firebase.iOS.MLKit" version="0.13.0.2" targetFramework="xamarinios10" />
  <package id="Xamarin.Firebase.iOS.MLKit.Common" version="0.13.0.1" targetFramework="xamarinios10" />

Looks like another update is required as 
https://github.com/firebase/quickstart-ios/issues/627

Sample Project
[TestMLUpgrade 3.zip](https://github.com/xamarin/GoogleApisForiOSComponents/files/4317229/TestMLUpgrade.3.zip)

### LINKED ISSUE #384: Wrong Binding in VisionFaceContour
For FIRVisionFaceContour, it should have FaceContourType but not FaceLandmarkType.
FYI
https://firebase.google.com/docs/reference/swift/firebasemlvision/api/reference/Type-Definitions.html#facecontourtype
```	
// @interface FIRVisionFaceContour : NSObject
	[DisableDefaultCtor]
	[BaseType (typeof (NSObject), Name = "FIRVisionFaceContour")]
	interface VisionFaceContour {
		// @property (readonly, nonatomic) FIRFaceContourType _Nonnull type;
		[BindAs (typeof (FaceLandmarkType))]
		[Export ("type")]
		NSString Type { get; }

		// @property (readonly, nonatomic) NSArray<FIRVisionPoint *> * _Nonnull points;
		[Export ("points")]
		VisionPoint [] Points { get; }
	}
```

### PR DESCRIPTION
**Pull request by SotoiGhost**
This update includes:
* A/B Testing 3.2.0
* AdMob 7.57.0
* Analytics 6.4.0
* Auth 6.5.1
* Core 6.6.5
* Cloud Firestore 1.11.2
* Cloud Messaging 4.3.0
* Crashlytics 4.0.0-beta.6
* Database 6.1.4
* Dynamic Links 4.0.8
* Installations 1.1.1
* Instance ID 4.3.2
* MLKit Common 0.19.0
* MLKit Model Interpreter 0.19.0
* MLKit Natural Language 0.17.0
* MLKit Vision 0.19.0
* Performance Monitoring 3.1.11
* Remote Config 4.4.9
* Storage 3.6.0

Also, this PR:
* fixes #319
* fixes #369 
* fixes #373 
* fixes #375 
* fixes #379 
* fixes #380 
* fixes #382 
* fixes #384

### CODE REVIEW

**Comment by SotoiGhost** [comment]
@dalexsoto Can you review these diffs? 🙏  Thanks in advance!
* Analytics
    * [ ] [Extensions](https://github.com/xamarin/GoogleApisForiOSComponents/pull/388/files?file-filters%5B%5D=.cs#diff-1bd233d4d46c95c085c57c61a8eae7b7)
* Auth
    * [ ] [ApiDefinition](https://github.com/xamarin/GoogleApisForiOSComponents/pull/388/files?file-filters%5B%5D=.cs#diff-e7deb5a574e7a87d5a289ac9087158f2)
    * [ ] [Enums](https://github.com/xamarin/GoogleApisForiOSComponents/pull/388/files?file-filters%5B%5D=.cs#diff-e2d0bdc14e191cc48457c4c1a935093e)
* Cloud Firestore
    * [ ] [ApiDefinition](https://github.com/xamarin/GoogleApisForiOSComponents/pull/388/files?file-filters%5B%5D=.cs#diff-5bf46b1f698d50143db305d540fc07eb)
* Cloud Messaging
    * [ ] [ApiDefinition](https://github.com/xamarin/GoogleApisForiOSComponents/pull/388/files?file-filters%5B%5D=.cs#diff-fb2cfbddb36a83ea67c4647b86007ebc)
* Crashlytics
    * [ ] [ApiDefinition](https://github.com/xamarin/GoogleApisForiOSComponents/pull/388/files?file-filters%5B%5D=.cs#diff-67a27ea01308c62253fb3457b51368e5)
    * [ ] [Extensions](https://github.com/xamarin/GoogleApisForiOSComponents/pull/388/files?file-filters%5B%5D=.cs#diff-ce3c5c6bd022aaf0e2e85599adfb530d)
* Installations
    * [ ] [ApiDefinitions](https://github.com/xamarin/GoogleApisForiOSComponents/pull/388/files?file-filters%5B%5D=.cs#diff-9f4837917800134e7a85574a8023589b)
    * [ ] [Enums](https://github.com/xamarin/GoogleApisForiOSComponents/pull/388/files?file-filters%5B%5D=.cs#diff-6fa24b924eeade5b3b1f06fc8ddfacdd)
    * [ ] [Extensions](https://github.com/xamarin/GoogleApisForiOSComponents/pull/388/files?file-filters%5B%5D=.cs#diff-9ff75a0042780d573434cd5139cdc738)
* MLKit Common
    * [ ] [ApiDefinition](https://github.com/xamarin/GoogleApisForiOSComponents/pull/388/files?file-filters%5B%5D=.cs#diff-0f49107915263c8f933dffb1576e7ed4)
* MLKit Model Interpreter
    * [ ] [ApiDefinition](https://github.com/xamarin/GoogleApisForiOSComponents/pull/388/files?file-filters%5B%5D=.cs#diff-7bb3aa2d6352d591f76ffb27d5dfcfbf)
    * [ ] [Extensions](https://github.com/xamarin/GoogleApisForiOSComponents/pull/388/files?file-filters%5B%5D=.cs#diff-170c205359dfc1269eaefe2e4199d2d9)
* MLKit Natural Language
    * [ ] [ApiDefinition](https://github.com/xamarin/GoogleApisForiOSComponents/pull/388/files?file-filters%5B%5D=.cs#diff-f56110d9fa1c2908d6eed73df143c029)
    * [ ] [Enums](https://github.com/xamarin/GoogleApisForiOSComponents/pull/388/files?file-filters%5B%5D=.cs#diff-c2026e454f42105aaaa4bffa9e3689d3)
* MLKit Vision: Clone the repo and checkout to `update/firebase/6.21.0`
    * [ ] ApiDefinition: `git difftool 7c4555f:source/Firebase/MLKit/ApiDefinition.cs efdb784:source/Firebase/MLKit.Vision/ApiDefinition.cs`
    * [ ] Enums: `git difftool 7c4555f:source/Firebase/MLKit/Enums.cs efdb784:source/Firebase/MLKit.Vision/Enums.cs`
* Remote Config
    * [ ] [ApiDefinition](https://github.com/xamarin/GoogleApisForiOSComponents/pull/388/files?file-filters%5B%5D=.cs#diff-ad447c8dca7810ebeac9be926bb7d6b5)
* Mobile Ads
    * [ ] [ApiDefinition](https://github.com/xamarin/GoogleApisForiOSComponents/pull/388/files?file-filters%5B%5D=.cs#diff-b0dd16c4406dbddc2d2bf51e15e166d6)

File: source/Firebase/MLKit.Common/ApiDefinition.cs [hunk 1]
```diff
@@ -6,60 +6,102 @@
...
+	[BaseType (typeof(NSObject), Name = "FIRLocalModel")]
+	interface LocalModel
+	{
+		// @property (readonly, copy, nonatomic) NSString * _Nonnull path;
+		[Export ("path")]
+		string Path { get; }
+	}
+
+	// @interface FIRModelDownloadConditions : NSObject <NSCopying>
+	[BaseType (typeof(NSObject), Name = "FIRModelDownloadConditions")]
```
Comment (dalexsoto) [inline source/Firebase/MLKit.Common/ApiDefinition.cs:20]:
```suggestion
	[BaseType (typeof(NSObject), Name = "FIRModelDownloadConditions")]
	[DisableDefaultCtor]
```

**Comment by SotoiGhost** [comment]
@dalexsoto Can you review the last one, please? 🙏  Thanks in advance!

* In-App Messaging
    * [ ] [ApiDefinition](https://github.com/xamarin/GoogleApisForiOSComponents/pull/388/files?file-filters%5B%5D=.cs#diff-581676bf639b7e217f034ea618bc0802)
    * [ ] [Enums](https://github.com/xamarin/GoogleApisForiOSComponents/pull/388/files?file-filters%5B%5D=.cs#diff-b8d7f9ee625c4e1a175fa3a72278da09)
    * [ ] [Extensions](https://github.com/xamarin/GoogleApisForiOSComponents/pull/388/files?file-filters%5B%5D=.cs#diff-db565df998e3ce426d2721bb2ba4010b)

**Comment by dogiaplinh** [comment]
Hey team, when will nuget packages be shipped?

**Comment by moustafashaban** [comment]
any update on new package?

**Comment by joshuabooker** [comment]
Hey guys, hope everything is well, 
any update on when the nuget package will be released? Having **FieldValue.Increment** seems alot better than creating a transaction to do the same thing.

**Pull request merged**